### ISSUE TITLE
Feature Request: Add Created and Modified Columns to Database Site Tables

### ISSUE THREAD

**Issue by playcock** [modxcms/revolution#13641]
### Summary
Database table columns to add: Created, Modified

### Step to reproduce
Creating/Saving a Resource or Element in the Manager

### Expected behavior
I would like to see created and a modified columns on the following database tables when creating or updating a Resource or Element from within the Manager, please. 

Both columns would be the timestamp type. 
The created column's Default would be CURRENT_TIMESTAMP
The modified column would contain the date and time whenever the Resource or Element was modified.
 
This would help me track when a resource or element is created and/or modified.

modx_site_content
modx_site_htmlsnippets
modx_site_plugins
modx_site_plugin_events
modx_site_snippets
modx_site_templates
modx_site_tmplvars
modx_site_tmplvar_access
modx_site_tmplvar_contentvalues
modx_site_tmplvar_templates

### Environment
MODX Revolution 2.5+

**Comment by playcock** [comment]
What I am looking for is a readable date.

**Comment by Mark-H** [comment]
For tracking modifications, the manager log is useful to look at. However I'd also like to see more created/edited timestamps and user IDs for accountability. 

That does beg the question of where those fields would be exposed to the user, otherwise they may have little value. Also what constitutes an edit - should interactions via xPDO cause the edited fields to be updated, or only though the manager?

**Comment by OptimusCrime** [comment]
I agree with updated timestamp, created timestamp, created by user and updated by user for snippets, templates, chunks and plugins. But I fail to see the use of this for the rest. Like @Mark-H already said, the manager log can be used to some extent for this.

**Comment by isaacniebeling** [comment]
`modx_site_content` already has timestamps... I don't think the others do (though `modx_site_tmplvar_contentvalues` could likely be assumed to be the same as `modx_site_content`). Probably related to #13626...

I'd say _elements_ make a lot of sense, but `modx_site_tmplvar_access` and `modx_site_tmplvar_templates` change along with `modx_site_tmplvars`, `modx_site_plugin_events` changes with `modx_site_plugins`... you _could_ manually change those via xpdo, I suppose, but... that seems like a super-duper-edgy-edge-case

**Comment by playcock** [comment]
I just like a peace of mind when pushing data from those tables from sandboxes to dev to staging to production with multiple people working on the web sites. The people working on the modx web sites could give me a list of resources, templates, snippets, and chunks that they have worked on, but we're people and make mistakes. 

So, I tag git deployments by versions and will be able to use the date of the last version and run a query on those tables as an extra measure to collaborate several CMS changes that need published.

If something doesn't seem right on the site, I can run queries on those site tables to find out if any of those tables were modified or created since the version release. I would probably run that query anyways, to make sure all intended content updated in the CMS gets released, because not everyone in the dev environments that work in the CMS are really technical and even technical people make mistakes.

I could trim down my request to these tables:
modx_site_content
modx_site_htmlsnippets
modx_site_snippets
modx_site_templates
modx_site_tmplvars
modx_site_tmplvar_contentvalues
modx_site_tmplvar_templates

**Comment by OptimusCrime** [comment]
The last two are redundant. Please explain when those could be different from the people that last edited the template variable itself, or the resource?

**Comment by playcock** [comment]
Are you saying that those could not be different from the people that last edited the template variable itself?

**Comment by OptimusCrime** [comment]
I can't see how? Unless you use the xPDO API, then all of these could be different, or impossible to track correctly.

**Comment by playcock** [comment]
That gives me some peace of mind.

**Comment by OptimusCrime** [comment]
If we can narrow it down to resources (add what is missing, if anything), templates, chunks, snippets, plugins and template variables, then I agree.

**Comment by playcock** [comment]
I agree with OptimusCrime. Any content (resources, templates, template vars, chunks, snippets[, plugins]) that is created or changed by the CMS user, should have their respective tables include a human readable created date and a human readable modified date. Anything more than that, is just bonus.

**Comment by OptimusCrime** [comment]
Good. Are you going to give this a go? I did not come off clearly if this is something you are currently working on?

**Comment by playcock** [comment]
No, no. I'm the new guy here and have plenty to get acclimated for first, I am sure. I originally posted this on the forums and was told that this repo was the correct place to post feature requests so that they do not get lost.

My post on the forums: https://forums.modx.com/thread/102990/feature-request-add-created-and-modified-columns-to-database-site-tables
